test-apps/appui-test-app/appui-test-providers/src/ui/frontstages/CustomFrontstageProvider.tsx

```diff
@@ -3,160 +3,46 @@
-
-class CustomContentControl extends ContentControl {
-  constructor(info: ConfigurableCreateInfo, options: any) {
-    super(info, options);
-
-    this.reactNode = (
-      <h1
-        style={{
-          height: "100%",
-          display: "flex",
-          alignItems: "center",
-          justifyContent: "center",
-        }}
-      >
-        Custom content!
-      </h1>
-    );
-  }
-}
-
-export class CustomFrontstageProvider extends FrontstageProvider {
-  public static readonly stageId = "appui-test-providers:CustomFrontstage";
-
-  public get id(): string {
-    return CustomFrontstageProvider.stageId;
+// __PUBLISH_EXTRACT_END__
+import {
+  ReactContentControl,
+  ReactContentControlOptions,
+} from "../content/ReactContentControl";
+
+// __PUBLISH_EXTRACT_START__ AppUI.FrontstageProvider
+class CustomFrontstageProvider extends FrontstageProvider {
+  public override get id(): string {
+    return "example:CustomFrontstage";
   }
 
   public override frontstageConfig(): FrontstageConfig {
-    const id = this.id;
     const contentGroup = new ContentGroup({
-      id: "test-group",
+      id: `${this.id}:content-group`,
       layout: StandardContentLayouts.singleView,
-      contents: [{ id: "custom-content", classId: CustomContentControl }],
+      contents: [
+        {
+          id: `${this.id}:content`,
+          classId: ReactContentControl,
+          applicationData: {
+            node: <h1>Custom Content</h1>,
+          } satisfies ReactContentControlOptions,
+        },
+      ],
     });
     return {
-      id,
+      id: this.id,
       version: 1,
       contentGroup,
-      contentManipulation: {
-        id: `${id}-contentManipulationTools`,
-        content: (
-          <ContentToolWidgetComposer
-            cornerButton={
-              <BackstageAppButton
-                label="Toggle Backstage"
-                icon="icon-bentley-systems"
-                execute={() =>
-                  UiFramework.backstage.getBackstageToggleCommand().execute()
-                }
-              />
-            }
-          />
-        ),
-      },
-      viewNavigation: {
-        id: `${id}-viewNavigationTools`,
-        content: <ViewToolWidgetComposer />,
-      },
-      toolSettings: {
-        id: `${id}-toolSettings`,
-      },
-      statusBar: {
-        id: `${id}-statusBar`,
-        content: (
-          <StatusBarComposer
-            key={UiFramework.frontstages.activeFrontstageId}
-            items={[]}
-          />
-        ),
-      },
-      leftPanel: {
-        size: 500,
-        defaultState: StagePanelState.Minimized,
-        sections: {
-          start: [
-            {
-              id: "widget-1",
-              label: "Widget 1",
-              content: (
-                <>
-                  Frontstage provided widget: <b>widget-1</b>
-                </>
-              ),
-            },
-          ],
-        },
-      },
     };
```

Comment:
Dont' we loose a lot of stuff here ? Is that intended?

Reply:
That was intentional, since it makes sense to provide minimal code snippets instead of huge code blocks that do everything.